services/packages/packages.go

```diff
@@ -26,9 +26,9 @@ import (
 )
 
 var (
-	ErrQuotaTypeSize   = errors.New("maximum allowed package type size exceeded")
-	ErrQuotaTotalSize  = errors.New("maximum allowed package storage quota exceeded")
-	ErrQuotaTotalCount = errors.New("maximum allowed package count exceeded")
+	ErrQuotaTypeSize   = util.NewInvalidArgumentErrorf("maximum allowed package type size exceeded")
+	ErrQuotaTotalSize  = util.NewInvalidArgumentErrorf("maximum allowed package storage quota exceeded")
+	ErrQuotaTotalCount = util.NewInvalidArgumentErrorf("maximum allowed package count exceeded")
```

Comment:
Is "you have reached a limit" an "argument error"?

Reply:
Nope. Was the closest type that was defined in the util package. Would it be better to expand util's error types?

Reply:
Would error `FailedPrecondition` (by analogy with gRPC status codes) fit this case?

Reply:
@KN4CK3R 

No. Added draft with `FailedPrecondition` error in [this branch](https://github.com/d1nch8g/gitea/tree/refactoring-package-upload-error-handle-failed-precondition). Would it be valid for current scenario?

Reply:
Is it a mistake that only `ErrQuotaTotalCount` is a `util.NewFailedPreconditionErrorf`?

Reply:
My bad, I was not sure about this error, changed to `util.ErrLimitExceeded`

Reply:
Yes, applied required corrections